src/main/scala/lms/transformation/distributed_tensor/distributedTensorConv.scala

```diff
@@ -0,0 +1,63 @@
+package lms.transformation.tensor
+
+import scala.annotation.implicitNotFound
+import scala.collection._
+
+import lms.core._
+import lms.core.stub._
+import lms.collection.mutable._
+import lms.macros.SourceContext
+import lms.thirdparty.array_computation.{ArrayCPUOps, CUDATypeLess, CudaOps}
+import lms.thirdparty.{CUDNNTypeLess, CUDNNOps}
+
+import Backend._
+
+
+trait FixedSizeDistributedTensorConvTypeLess extends FixedSizeDistributedTensorMutationTypeLess {
+
+  def ConvBackwardData(weight: TENSOR, filter: TENSOR, doutput: TENSOR, alpha: Float, beta: Float, 
+                      padding: Seq[Int], strides: Seq[Int], dilation: Seq[Int],
+                      anno: Anno, __pos: SourceContext): TENSOR = {
+    assert(weight.et == filter.et && filter.et == doutput.et)
+    val res_tt = weight.tensor_type
+    (new TENSOR(Adapter.g.reflectRead("tensor_conv_bwd_data", C(res_tt), C(anno), 
+      filter.x, doutput.x, C(alpha), C(beta, C(padding), C(strides), C(dilation)))
```

Comment:
missing a parameter after `C(beta`?

I am think whether it makes sense to create a `class ConvParam` that includes `pad`, `strides`, and all related parameters of Conv ops. Then in the IR, instead of having 4 C(xxx), we just have 1 C(p: ConvParam).

Reply:
Yeah we could do that.

Reply:
parenthesis fixed: fdf071b